app/src/main/java/ch/epfl/polybazaar/map/MapsActivity.java

```diff
@@ -0,0 +1,208 @@
+package ch.epfl.polybazaar.map;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.location.Location;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.ImageView;
+import android.widget.Toast;
+
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.android.gms.tasks.Task;
+
+import java.util.Objects;
+
+import ch.epfl.polybazaar.R;
+import ch.epfl.polybazaar.database.callback.SuccessCallback;
+import ch.epfl.polybazaar.widgets.permissions.PermissionRequest;
+
+import static ch.epfl.polybazaar.map.Constants.*;
+
+public class MapsActivity extends AppCompatActivity implements OnMapReadyCallback, GoogleMap.OnMyLocationButtonClickListener {
+
+    private GoogleMap mMap;
+
+    private final String TAG = "MapsActivity";
+
+    private FusedLocationProviderClient mFusedLocationProviderClient;
+    private PermissionRequest perm;
+    private boolean mLocationPermissionGranted = false;
+    private boolean MPSet = false;
+
+    // in showMode, the user cannot select a meeting point
+    private boolean showMode = true;
+    private Bundle extras;
+    private ImageView imgMyLocation;
+
+    @Override
+    public void startActivityForResult(Intent intent, int requestCode) {
+        super.startActivityForResult(intent, requestCode);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_maps);
+        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
+        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager()
+                .findFragmentById(R.id.map);
+        mFusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(this);
+        assert mapFragment != null;

```

Comment:
I am not sure `assert` is the correct way to handle errors here. Maybe we should have a generic "Something unexpected happened" dialog for the app ? This might be useful for many different things.

Reply:
Yes, I must have missed this assertion, I'll fix that.